repository-ydb-v1/src/test/java/tech/ydb/yoj/repository/ydb/YdbRepositoryIntegrationTest.java

```diff
@@ -863,6 +866,30 @@ public void schemaWithHint() {
         repository.schema(HintAutoPartitioningByLoad.class).create();
     }
 
+
+    @Test
+    public void doNotCommitAfterTLI() {
+        Project.Id id1 = new Project.Id("id1");
+        Project.Id id2 = new Project.Id("id2");
+
+        RepositoryTransaction tx = repository.startTransaction(
+                TxOptions.create(IsolationLevel.SERIALIZABLE_READ_WRITE)
+                        .withImmediateWrites(true)
+                        .withFirstLevelCache(false)
+        );
+
+        tx.table(Project.class).find(id2);
+
+        db.tx(() -> db.projects().save(new Project(id2, "name2")));
+
+        tx.table(Project.class).save(new Project(id1, "name1")); // make tx available for TLI
+
+        assertThatExceptionOfType(OptimisticLockException.class)
+                .isThrownBy(() -> tx.table(Project.class).find(id2));
+
+        tx.commit(); // Commit shouldn't throw an BadSession or other exceptions
```

Comment:
Is it even correct to call `commit()` on a transaction *that got invalidated* (allowing no-error `rollback()` I can kinda understand)? 
I think trying to commit an invalidated transaction should definitely cause `IllegalStateException` or something like that, no?

Reply:
In this PR, I wanted to touch the old logic to a minimum. commit() is controlled from TxManger, it cannot make a commit after TLI or BadSession, but maybe someone other can. To reduce the likelihood of strange illegals falling out, I did not check it here. This is what this test shows.
But still, it’s really more correct to throw an exception here, I’ll do it.

Also I have a proposal-PR which make works with session and exceptions more understandable (maybe), but I pull it after this.